refactor: adjust PopularSection Pager sizing

Noor-Serry opened the pull request:
The HorizontalPager in PopularSection now uses `PageSize.Fixed` with a width of 244.dp. This ensures consistent item sizing.

The `itemWidth` and `horizontalPadding` are now remembered to avoid recalculations.

The PopularMediaItemCard now uses `fillMaxSize()` to ensure it correctly occupies the space allocated by the pager.

# Pull Request Template

## Description
Provide a clear and concise description of what this pull request does. Include the purpose and context for the changes.
now its more similar to the design its not 100 % accurate but its more similar now the padding between the center item and 
the item beside it is 18.5 and this 18.5  and in the design is 16.dp
but now the center item width is 244.dp 
---

## Changes Made
List the changes introduced in this pull request:



---

## Screenshots (if applicable)
Include screenshots or GIFs to showcase the changes, especially if they impact the UI.

https://github.com/user-attachments/assets/2b8355b2-9a57-41f3-9ef6-e3c22324a277

---

## Checklist
Please ensure the following tasks are completed:

- [x] My code follows the code style of this project
- [x] Changes have been tested manually and verified.
- [x] PR includes at most one single feature.

---

## Additional Comments
Add any additional information or context about the pull request here.

Noor-Serry committed:
refactor: adjust PopularSection Pager sizing

The HorizontalPager in PopularSection now uses `PageSize.Fixed` with a
width of 244.dp. This ensures consistent item sizing.

The `itemWidth` and `horizontalPadding` are now remembered to avoid
recalculations.

The PopularMediaItemCard now uses `fillMaxSize()` to ensure it correctly
occupies the space allocated by the pager.

ThoriaHamdy99 reviewed (commented):
What is it look like in landscape, can you provide a photo?

Noor-Serry commented:
@ThoriaHamdy99 
as I mention in the daily meeting in  landscape the space between the second item and the third item is 37 
but if you have any solution please  mention it
https://github.com/user-attachments/assets/abf52237-a583-47ad-b4bb-9c94ba9a6df2

nabilBouzineDev commented on ui/src/main/java/com/amsterdam/ui/screens/home/sections/PopularSection.kt:
```diff
@@ -188,11 +189,11 @@ private fun PopularMoviesPager(
-    val itemWidth = 207.dp
-    val horizontalPadding = (screenWidth - itemWidth) / 2
+    val itemWidth = remember { 244.dp }
+    val horizontalPadding = remember { (screenWidth - itemWidth) / 2 }
```
Add a check to ensure this calculation will never be negative because that may cause an exception at run time since padding values should not be negative.

Noor-Serry replied:
that is right

Noor-Serry committed:
fix: prevent negative horizontal padding in PopularSection Pager

The horizontal padding for the `HorizontalPager` in `PopularSection` is now coerced to be at least `0.dp`. This prevents potential negative padding values when the `itemWidth` is larger than the `screenWidth`, ensuring correct layout behavior.
